core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala

```diff
@@ -1056,10 +1056,27 @@ class PlaintextAdminIntegrationTest extends BaseAdminIntegrationTest {
           assertTrue(latch.await(30000, TimeUnit.MILLISECONDS))
           // Test that we can list the new group.
           TestUtils.waitUntilTrue(() => {
-            val matching = client.listConsumerGroups.all.get().asScala.filter(_.groupId == testGroupId)
-            matching.nonEmpty
+            val matching = client.listConsumerGroups.all.get.asScala.filter(group =>
+                group.groupId == testGroupId &&
+                group.state == ConsumerGroupState.UNKNOWN)
```

Comment:
I find the usage of `UNKNOWN` here is not intuitive because, at first, it gives the impression that a group can be in this state. I tend to think that using an Optional in `ConsumerGroupListing` would be better when we actually don't have the information. What do you think?

Reply:
Yes that's a good idea! 👍